vio/viosslfactories.c

```diff
@@ -26,7 +26,12 @@
 static my_bool     ssl_algorithms_added    = FALSE;
 static my_bool     ssl_error_strings_loaded= FALSE;
 
-/* the function below was generated with "openssl dhparam -2 -C 2048" */
+#if !defined(WOLFSSL) && !defined(SSL_CTRL_SET_DH_AUTO)
```

Comment:
It appears that `#define SSL_CTRL_SET_DH_AUTO` was added considerably before OpenSSL v1.0.2: https://github.com/openssl/openssl/commit/09599b52d4e295c380512ba39958a11994d63401 ("Auto DH support."), first tagged in https://github.com/openssl/openssl/releases/tag/OpenSSL_1_1_0-pre1.

Will this feature _actually work_ with OpenSSL >=1.1.0-pre1 < 1.0.2?

Should there additionally be a check for `OPENSSL_VERSION_NUMBER < 0x10200000L`?